components/Payment/Payment.tsx

```diff
@@ -20,18 +21,30 @@ function Payment(props: Props) {
 
     function onPayPaypal(productId: string, coflCoins?: number) {
         setLoadingId(coflCoins ? `${productId}_${coflCoins}` : productId)
-        api.paypalPurchase(productId, coflCoins).then(data => {
-            setCurrentRedirectLink(data.directLink)
-            window.open(data.directLink)
-        })
+        api.paypalPurchase(productId, coflCoins)
+            .then(data => {
+                setCurrentRedirectLink(data.directLink)
+                window.open(data.directLink)
+            })
+            .catch(() => {
+                setCurrentRedirectLink('')
+                setLoadingId('')
+                toast.error('Something went wrong. Please try again.')
```

Comment:
any mention of traceId or the error message if its not "internal_error"?

Reply:
also, redundant code

Reply:
> any mention of traceId or the error message if its not "internal_error"?

If I get a error object from the backend, I display the message. This is the default error handling in the apihelper.
![image](https://user-images.githubusercontent.com/58751503/175357854-a403d7d5-c4cf-438e-be0d-edfff9544c9b.png)